### PR TITLE
Feat: use ruff as python linter!

### PR DESCRIPTION
**Pull request by hydrielax**
# Description

Replace Flake8 and Black by Ruff!

Ruff is a linter and formatter coded in Rust, allowing real-time warnings (no need to save the file to see the errors!), proposes fixes for most common errors, showing a link with a full description of the problem,
and a far better integration into vscode!

It also now catches more useful errors, and removed previous errors that were not useful.

### CODE REVIEW

**Comment by hydrielax** [comment]
@rravelli Need your help: I don't understand why the test for resetting the password do not pass

**Comment by rravelli** [comment]
> @rravelli Need your help: I don't understand why the test for resetting the password do not pass

Je check ça...

File: backend/apps/account/apps.py [hunk 1]
```diff
@@ -5,6 +5,4 @@ class AccountConfig(AppConfig):
     name = "apps.account"
 
     def ready(self) -> None:
-        import apps.account.signals  # noqa: WPS433,F401
```
[the remark this answers is not in the document]
Comment (hydrielax) [reply backend/apps/account/apps.py:8]: Ok je fix ça, merci !

[the remark this answers is not in the document]
Comment (hydrielax) [reply backend/apps/account/apps.py:8]: Ok en fait j'ai plusieurs bouts de code qui ont été supprimés alors qu'ils auraient pas dû l'être, je relis tout ça

**Review by rravelli — APPROVED** [review]
Nice, enfin une config python qui tient la route ! Aurevoir les # noqa :hand:

**Pull request merged**